armi/utils/tests/test_gridGui.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2021 Google, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for gridEditor.py.
+
+By default, this would open the app in your primary display and steal control from your mouse and keyboard while the
+test is running. This means that if your display is smaller than 1000 x 1000, or if you move your mouse when the test is
+running, the test might fail even when things are fine.
+
+These issues can be resolved by running the test in a virtual display. To do so, `pip install pytest-xvfb`,
+and run `pytest test_gridGui.py` as usual. If you wish to change the resolution of the virtual display, you can modify
+the `xvfb_width` and `xvfb_height` in `pytest.ini`.
+
+After you have installed pytest-xvfb, you will no longer see the app displayed on your screen, which can make debugging
+harder. Thus, during debugging, you may want the app to appear on your display by setting the `--no-xvfb` flag, and
+have the print statements print to your console by setting the `--capture=tee-sys` flag, like this:
+```
+pytest --no-xvfb --capture=tee-sys test_gridGui.py
```

Comment:
In order to get the tests to run, I had to be in a pure X11 environment; default Gnome on my system runs on Wayland with xwayland to support X clients, which didn't work. Now that I think about it, the more security-minded nature of Wayland may even make this sort of GUI testing impossible? I'm not sure... anyways, it might be worth noting known requirements on the test environment.

Reply:
Thanks for figuring this out! I've noted this in the comment.